eng/pipelines/spec-gen-sdk-all-specs.yml

```diff
@@ -0,0 +1,157 @@
+parameters:
+  - name: RunMode
+    type: string
+    values:
+      - 'all-specs'
+      - 'all-typespecs'
+      - 'all-openapis'
+      - 'sample-typespecs'
+    default: 'sample-typespecs'
+    displayName: 'Specs to Run'
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages:
+    - stage: Build
+      displayName: 'SDK Generation'
+      jobs:
+      - job:
+        displayName: 'SDK Generation'
+
+        variables:
+          - template: /eng/pipelines/templates/variables/image.yml
+          - name: NodeVersion
+            value: '22.13.x'
+          - name: PythonVersion
+            value: '3.13'
+          - name: SpecGenSdkVersion
+            value: 'latest'
+          - name: SdkArtifactName
+            value: SDK_Artifact
+          - name: SpecRepoUrl
+            value: 'https://github.com/$(Build.Repository.Name)'
+
+        pool:
+          name: $(LINUXPOOL)
+          vmImage: $(LINUXVMIMAGE)
+          os: linux
+
+        templateContext:
+          outputParentDirectory: $(System.DefaultWorkingDirectory)/out
+          outputs:
+            - output: pipelineArtifact
+              displayName: Publish SDK artifacts to Pipeline Artifacts
+              condition: and(ne(variables['ValidationResult'], ''), eq(variables['HasSDKArtifact'], 'true'))
+              artifactName: $(sdkArtifactName)
+              targetPath: "$(System.DefaultWorkingDirectory)/out/generatedSdkArtifacts"
+            - output: pipelineArtifact
+              displayName: Publish API View artifacts to Pipeline Artifacts
+              condition: and(ne(variables['ValidationResult'], ''), eq(variables['HasApiViewArtifact'], 'true'))
+              artifactName: $(ArtifactName)
+              targetPath: "$(System.DefaultWorkingDirectory)/out/sdkApiViewArtifacts"
+            - output: pipelineArtifact
+              displayName: Publish logs to Pipeline Artifacts
+              condition: ne(variables['ValidationResult'], '')
+              artifactName: "spec-gen-sdk-logs"
+              targetPath: "$(System.DefaultWorkingDirectory)/out/logs"
+
+        steps:
+          - checkout: none
+
+          - pwsh: |
```

Comment:
This looks like a lot of duplication from the eng/common pipeline Is there a reason we cannot share?

Reply:
That's correct. My plan is to consolidate similar steps into a common template for sdk generation, with two entry pipeline YAML files extending from it, one is the `spec-gen-sdk` and the other is the `spec-gen-sdk-all-specs`.
To streamline the process, the step calling `spec-gen-sdk` will be updated to use `/eng/tools/gen-sdk/cmd/gen-sdk.js`, ensuring consistency across all scenarios, including release, spec PRs, and spec-gen-sdk-all-specs, etc.
Additionally, the logic for computing related files, iterating over spec configuration files, and handling pre-processing before passing the pure spec config path to spec-gen-sdk will all be encapsulated within the wrapper tool at `eng/tools/gen-sdk`.

Does this approach align with your expectations?

Reply:
OK that mostly aligns with my expectations. I would like almost everything in the yml file to be shared, I think there should only be a couple differences between check PR vs check set of specs.

As for where the wrapper tool resides, I would say it should stay in the specs repo under eng/tools given it will only be used by these PRs in the specs repo and nowhere else.

Reply:
I don't believe it will be used by the SDK language repos, and I agree that it makes sense to move it to spec repo.